hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -727,6 +728,11 @@ public class HoodieWriteConfig extends HoodieConfig {
    */
   public static final String SPARK_SQL_MERGE_INTO_PREPPED_KEY = "_hoodie.spark.sql.merge.into.prepped";
 
+  /**
+   * An internal config referring to fileID encoding. 0 refers to UUID based encoding and 1 refers to raw string format(random string).
+   */
+  public static final String WRITES_FILEID_ENCODING =  "_hoodie.writes.fileid.encoding";
```

Comment:
NTR: instead of introducing an internal config, we could intercept the fileID format just before constructing RLI records and we auto deduct encoding. but at scale, parsing the fileID to deduce UUID format might have some perf hit. So, going via internal config might be a better option. 

Open to ideas if any.